src/main/java/jenkins/plugins/logstash/LogstashConfiguration.java

```diff
@@ -53,6 +58,38 @@ public void setEnableGlobally(boolean enableGlobally)
     this.enableGlobally = enableGlobally;
   }
 
+  public boolean isMilliSecondTimestamps()
+  {
+    return milliSecondTimestamps;
+  }
+
+  public void setMilliSecondTimestamps(boolean milliSecondTimestamps)
+  {
+    this.milliSecondTimestamps = milliSecondTimestamps;
+    createDateFormatter();
+  }
+
+  public FastDateFormat getDateFormatter()
+  {
+    if (dateFormatter == null)
+    {
+      createDateFormatter();
+    }
+    return dateFormatter;
+  }
+
+  private void createDateFormatter()
```

Comment:
Actually wouldn't it be simpler to create the 2 instances as constants instead of lazy?
`getDateFormatter()` would look sth like

```
 return milliSecondTimestamps? MILIS_FORMATTER : LEGACY_FORMATTER
```
```

Reply:
agreed